src/main/java/com/limechain/trie/structure/BranchNodeHandle.java

```diff
@@ -0,0 +1,41 @@
+package com.limechain.trie.structure;
+
+/**
+ * Has the semantics of a "branch node with no storage value",
+ * i.e. in the purely "structural necessity branch node" sense.
+ */
+public final class BranchNodeHandle<T> extends NodeHandle<T> {
+
+    // NOTE: An ugly workaround to simulate the notion of "consuming" this branch node handle
+    private boolean consumed;
+    BranchNodeHandle(TrieStructure<T> trieStructure, int nodeIndex) {
+        super(trieStructure, nodeIndex);
+        consumed = false;
+    }
```

Comment:
Is this intentionally left as package-private?

Reply:
> Is this intentionally left as package-private?

Yes, since only the trie structure can construct its handles, whilst they're intended to be used outside by the user.
For the detailed clarification of why some things are package-private: [reference](https://github.com/LimeChain/Fruzhin/pull/265#discussion_r1442929751).